client/packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -44,10 +44,14 @@ const DataTableComponent = <T extends RecordWithId>({
   const { setRows, setDisabledRows, setFocus } = useTableStore();
   const [clickFocusedRow, setClickFocusedRow] = useState(false);
   const [displayColumns, setDisplayColumns] = useState(columns);
-  const columnsToDisplay = columns.filter(c =>
-    displayColumns.map(({ key }) => key).includes(c.key)
+  const columnsToDisplay = React.useMemo(
```

Comment:
this one too - the `columnsToDisplay` changed on each render, which meant that the whole component had to re-render